seive.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+function seiveOriginal(n) {
+  let primes = new Array(n - 1).fill(true);
+  [primes[0], primes[1]] = [false, false];
+  for (let i = 2; i <= n; i++) {
+    for (let j = i * i; j <= n; j += i) {
```

Comment:
`primes` array consists of `n-1` elements but `j` can have `n-1` and `n` values according to this loop. It doesn't fail because it's JavaScript :man_shrugging: but in it would be a segmentation error in C and IndexOutOfBoundsException in Java.